packages/diagrams/backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/api/IToolService.java

```diff
@@ -27,6 +30,11 @@ public interface IToolService {
 
     Optional<ITool> findToolById(IEditingContext editingContext, Diagram diagram, String toolId);
 
+    /**
+     * Used to compute and add variables providing by the tool from the graphQL API to the {@link VariableManager}.
+     */
+    void addToolVariablesInVariableManager(List<ToolVariable> toolvariables, IEditingContext editingContext, VariableManager variableManager);
+
```

Comment:
Why would this code be here?

Reply:
The front can now return some variables. I thought it would have been interesting to have it in this service.